### DISCUSSION TITLE
Tracking optimum for other variables

### DISCUSSION THREAD

**Discussion by Atulsingh92** [anyoptimization/pymoo#255] [Q&A] [answered]
Hello, 

I have set a variable as `out["kpi"]` to store all my properties (temperature velocities, pressure etc), that are important other than `out["F"] or "G"`.
And I want to plot them to see how far they are from the constraint value visually.  For this I have used the `opt.get()` method as
`opt_kpi1  = np.array([e.opt.get("kpi")[:,11] for e in res.history])`

as my constraint boundary is lets say 1.05, the plot looks as follows,  (x is n_evals) calculated from [here](https://pymoo.org/misc/convergence.html?highlight=n_evals)

![image](https://user-images.githubusercontent.com/30570817/155907160-f4d47f82-51d5-4f76-8151-2e29d6c48f51.png)

Am I correct to assume this is tracking the convergence of the optimum, for my kpi value? And as the iterations progress, the results become more and more feasible? If that is the case, my results picks up this constraint as 

```
G_indices = [0dim1, kpi1, kpi2, kpi3 kpi4]
[    -0.17        -0.018        2.95918   -842.0915     349.7911]
```
So visually green-yellow , if negative is constraint satisfied, and if not, is a constraint violation. Is that correctly shown here?

Basically needed more insight on this `"opt" ` object, as the page says "The algorithm object has the attribute opt (a Population object), which contains the current optimum." So is my way of depiciting this `-0.17` value correct in the shown plot?

**Comment by blankjul** [comment, marked as answer]
Your code for tracking features of the current optimum looks good to me! `opt` always is the best solution found so far across all algorithms. Keep in mind that this could also be the least infeasible solution. If any feasible solution has been found throughout the search yet, it will be the one with the lowest function value (or the whole non-dominated set of solutions).

The interpretation of these results is application dependent.
